MantidPlot/src/Mantid/InstrumentWidget/InstrumentWindow.cpp

```diff
@@ -848,32 +69,31 @@ void InstrumentWindow::preDeleteHandle(
 void InstrumentWindow::afterReplaceHandle(
     const std::string &wsName, const boost::shared_ptr<Workspace> workspace) {
   // Replace current workspace
-  if (wsName == m_workspaceName.toStdString()) {
-    if (m_instrumentActor) {
+  if (wsName == m_instrumentWidget->getWorkspaceName().toStdString()) {
```

Comment:
The entire of this [method](https://github.com/mantidproject/mantid/blob/14911_Refactor_Instrument_View_Code/MantidPlot/src/Mantid/InstrumentWidget/InstrumentWindow.cpp#L72-L97) should be handled by the `InstrumentWidget` since the `InstrumentWidget` holds/owns the workspace, you can greatly increase encapsulation by pushing this functionality into the InstrumentWidget. call the method something like `InstrumentWidget::handleWorkspaceReplacement` and let it take the same parameters as the enclosing `afterReplaceHandle`. While you're working on this, try to push things like the same workspace determination into a free function or private member function on the `InstrumentWidget`